src/Carbon/CarbonInterval.php

```diff
@@ -596,6 +605,36 @@ public function __toString()
         return $this->forHumans();
     }
 
+    /**
+     * Convert the interval to a DatePeriod in given period.
+     *
+     * @param \DateTime|\DateTimeInterface     $start
+     * @param \DateTime|\DateTimeInterface|int $end
+     * @param int                              $options
+     *
+     * @throws \InvalidArgumentException
+     *
+     * @return \DatePeriod
+     */
+    public function toPeriod($start, $end, $options = 0)
+    {
+        if ($this->spec() == 'PT0S') {
+            throw new InvalidArgumentException('Empty interval cannot be converted into a period.');
+        }
+
+        if (!($options & static::EXCLUDE_END) && ($end instanceof DateTime || $end instanceof DateTimeInterface)) {
+            $end = clone $end;
+            $end = $end->modify('+1 second');
```

Comment:
modify is not part of the DateTimeInterface. So we can either use `$end instanceof DateTime || $end instanceof DateTimeImmutable` could restrict some custom classes implementing DateTimeInterface and a compatible modify method. Or we can use `method_exists($end, 'modify')`, it's the most flexible but we have no guarantee of what this modify method can be but as DatePeriod constructor take DateTime|DateTimeInterface|int, it should not cause unexpected behavior.

Last option: fallback to `getTimestamp ` method that exists in DateTimeInterface and can allow to create a new object with 1 more second.

Reply:
`DateTimeInterface` can only be implemented by built in classes so changing it to `DateTimeImmutable` will be mostly cosmetic :)

Last option you suggest would be `$end = Carbon::createFromTimestamp($end->getTimestamp(), $end->getTimezone());`, is that right?

Reply:
Yep, changing DateTimeInterface to DateTimeImmutable will be closest semantically, it means we handle both DateTime and DateTimeImmutable, because they both handle modify in known way, but for any future DateTimeInterface classes that can be created in PHP, as modify is not required by this interface, we cannot guarantee it will still work. In this particular case, I would prefer to have `$end instanceof DateTime || $end instanceof DateTimeImmutable`.

I also realize than with EXCLUDE_END, we get a object with `$start` referenced and `$end` copied. I have to do some check to see if it's reliable enough.

Reply:
At least on PHP 7.2.3 start and end instances seemed to be cloned when instantiating `DatePeriod`.